source/simulator/assembly.cc

```diff
@@ -1624,14 +1644,37 @@ namespace aspect
     // TODO: Compute artificial viscosity once per timestep instead of each time
     // temperature system is assembled (as this might happen more than once per
     // timestep for iterative solvers)
-    const double nu
-      = compute_viscosity (scratch,
+
+    double nu = 0;
+	int count_additions = 0;
+	double nu_additions = 0.0;
+    if (use_smoothing == false)
+    {
+      nu = compute_viscosity (scratch,
                            global_max_velocity,
                            global_field_range.second - global_field_range.first,
                            0.5 * (global_field_range.second + global_field_range.first),
                            global_entropy_variation,
                            cell->diameter(),
                            advection_field);
+    }
+    else
+    {
+    	for (int k=0;k<4;k++) // RG: all of this won't work if 5 neighbors
+    	{
+    		if(cell->neighbor_index(k)!=-1) // RG: is there a neighbor function?
+    		{
+    			nu_additions += viscosity_per_cell[cell->neighbor(k)->user_index()];
+    			count_additions += 1;
+    		}
+    	}
+    	if (count_additions != 0)
+    	{
+    		nu = 0.5*nu;
+    		nu += 0.5 * nu_additions/double(count_additions);
```

Comment:
Are you taking the average of the value on the current cell and the average over the neighbors? Would be worth adding a comment. I suspect that that is not a good strategy. I would just take the maximum of this cell and all neighbors.